test/functional/lua/fs_spec.lua

```diff
@@ -78,6 +78,17 @@ describe('vim.fs', function()
         return vim.fs.find(nvim, { path = dir, type = 'file' })
       ]], test_build_dir, nvim_prog_basename))
     end)
+    it('works with filter()', function()
+      eq(vim.fn.expand('%'), exec_lua([[
+        local dir = ...
+        return vim.fs.find('fs_spec.lua', {
+          path = dir, type = 'file',
+          filter = function (path)
+            return path:match('test$') or path:find('test/functional.*')
+          end
+        })
+      ]], nvim_dir))
+    end)
```

Comment:
IDK if this is correct or not so i might need help with testing :)

Reply:
`TEST_FILE=test/functional/lua/fs_spec.lua make functionaltest` to run the test. This test currently does not work (you cannot use `vim.fn` in the test function).